cmip6_downscaling/disagg/terraclimate.py

```diff
@@ -262,160 +222,252 @@ def hydromod(t_mean, ppt, pet, awc, soil, snow_storage, mfsnow):
         drain_soil = delta_soil * (1 - np.exp(-soil / awc))
     else:
         drain_soil = 0
 
-    demand = pet
+    # In this model aet is calculated based upon a demand/supply
+    # relationship where demand is PET and supply is the sum of
+    # input_h2o and snow_drink (with input_h2o being 95% of the
+    # runoff from melt+rain and snow_drink is like a sublimation term)
     supply = input_h2o + snow_drink
 
-    if demand >= supply:
+    if pet >= supply:
         # if there is more evaporative demand than supply,
         # aet will be constrained by the soil drainage
         # and your deficit will be the difference between
         # pet and aet
         aet = supply - drain_soil
         deficit = pet - aet
         runoff = 0
-        soil = soil + drain_soil
+        soil = soil + drain_soil  # remove from soil
     else:
         # if there is enough water to satisfy pet then
         # aet will match pet and there is no deficit
         aet = pet
         deficit = 0
 
-    # this boolean is hardcoded true right now
-    if snow_drives_hydrology:
-        excess = max(0, soil + delta_soil - awc)
-        excess_rain_only = max(0, soil + excess_after_liquid - awc)
-        if (demand < supply) and (soil + delta_soil > awc):
+        if (soil + delta_soil) > awc:
             # if supply exceeds demand and the updated soil water is greater than
             # the available capacity
-            runoff = excess
-            runoff_snow += excess - excess_rain_only
+            runoff = max(0, soil + delta_soil - awc)
+            runoff_snow = min(extra_runoff, melt)
+
+            rain_input = rain - extra_runoff - runoff_snow
+            excess_after_liquid = max(0, rain_input - pet)
+            excess_rain_only = max(0, soil + excess_after_liquid - awc)
+            runoff_snow += runoff - excess_rain_only
             soil = awc
-        elif (demand < supply) and (soil + delta_soil <= awc):
+        else:
             # if supply exceeds demand and updated soil water
-            # is less than available water capacity
+            # is less than available water capacity you update
+            # the soil by the change in soil
             soil += delta_soil
             runoff = 0
 
+    # add the extra runoff component
+    runoff += extra_runoff
+
     return {
         'aet': aet,
-        'deficit': deficit,
-        'runoff': runoff,
-        'snow_storage': snow_storage,
+        'def': deficit,
+        'q': runoff,
+        'swe': swe,
         'soil': soil,
-        'runoff_snow': runoff_snow,
     }
 
 
-def aetmod(et0, h2o_input, awc, soil_prev=None):
-    """Calculate monthly actual evapotranspiration (AET)
+def pdsi(
+    ppt: pd.Series,
+    pet: pd.Series,
+    awc: float,
+    pad_years: int = 10,
+    y1: int = CLIMATE_NORMAL_PERIOD[0],
+    y2: int = CLIMATE_NORMAL_PERIOD[1],
+) -> np.ndarray:
+    """Calculate the Palmer Drought Severity Index (PDSI)
```

Comment:
add a note at the top here that states this is just a wrapper for the climate_indicies implementation. Also point to @orianac's issue trying to reproduce John's implementation.